Proposed Support for Vue Multi File Components, Typescript Support and updated dependencies

rfruesmer opened the pull request:
Hi, 

I've updated the 3rd party dependencies to most recent versions and added support for Vue Multi File Components and mixing of TypeScript/JavaScript as was needed for our current projects.

Not 100% sure, but I guess we don't need the dependency to the konan package anymore with the changes that I've made, please check this.

And please be aware that before publishing a new version to the npm registry, we need to bump major versions on this module-structure plugin and the module-structure main repo as well.

WIStudent commented:
I am not exactly sure how your .vue files with typescript look like.

This is the `sample.vue` file, that is currently used in the test:
```vue
<template>
    <div></div>
</template>
<script>
  import moduleA from './module-a';
  import moduleB from './module-b.vue';
  const moduleC = require('./module-c.js');
  import moduleD from './module-d';
  import moduleE from 'aliasModuleE';
  import moduleF from 'aliasModuleF';
  import {parse} from '@vue/component-compiler-utils';
  const fs = require('fs');
</script>
```
The test expects these dependencies:
```js
[
  'module-a.js',
  'module-b.vue',
  'module-c.js',
  'module-d/index.js',
  'module-e.js',
  'module-f.vue'
]
```
Using the master branch, the test succeeds for `<script>`, `<script lang="js">` and `<script lang="ts">`. If I try the test with this pull reqest, it fails for `<script lang="js">` and `<script lang="ts">`.

rfruesmer commented:
Fixed the problems we discussed on MS teams. The konan module isn't able to parse inline TypeScript code if there's more than just import statements, e. g. TS-like class declarations it fails - so I had to introduce a TypeScript parser to handle such cases. Also updated the test cases to check most common scenarios.
